### PR TITLE
OCPBUGS-15723: Let getMachinesForNodePool return machines ordered by creation Timestamp

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
This prevent the NodePool controller from producing a different condition message to convey the same state of the world

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 1]
```diff
@@ -2548,3 +2548,25 @@ func (r *NodePoolReconciler) getPullSecretName(ctx context.Context, hostedCluste
...
+func (o machinesByCreationTimestamp) Less(i, j int) bool {
+	if o[i].CreationTimestamp.Equal(&o[j].CreationTimestamp) {
+		return o[i].Name < o[j].Name
+	}
+	return o[i].CreationTimestamp.Before(&o[j].CreationTimestamp)
+}
+
+// SortedByCreationTimestamp returns the machines sorted by creation timestamp.
+func sortedByCreationTimestamp(machines []*capiv1.Machine) []*capiv1.Machine {
+	res := make(machinesByCreationTimestamp, 0, len(machines))
```
Comment (muraee) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:2566]: this function can be simplified:
```
sort.Sort(machinesByCreationTimestamp(machines))
return machines
```
or is there a reason you want to keep the original slice unmodified?

**Comment by csrwng** [comment]
lgtm apart from @muraee 's comment

**Pull request merged**